File: FreqChar.c
/*
Write a program accept string from user and accept one character.
Return frequency of that character.
input: Marvellous
        M
output: 1
*/
#include<stdio.h>


int FreqChar(char *str,char ch)
{
    int count=0;
 while(*str!='\0')
    {
        if(*str == ch)
        {
          count++;
        }
        str++;
    }
    return count;
}
int main()
{
    char arr[20];
    char cValue;
    int bRet;

    printf("Enter String:\n");
    scanf("%[^'\n']s",arr);

    printf("Enter the character:\n");
    scanf("%c", &cValue);

    bRet = FreqChar(arr,cValue);
    printf("Character frequency %d",bRet);  
    return 0;
}

File: FreqChar.java
/*
    Accept Array of character from user and accept one character.
    Return difference between freq of capital and freq of small chaaracter

*/

import java.util.*;
class FreqChar
{
    public static int FreqCharacter(char Arr[])
    {
        int iCount1 = 0,iCount2 = 0;

        for(int i = 0; i < Arr.length; i++)
        {
            if((ch == 'a') || (ch == 'z'))
            {
                if(Arr[i] == ch)
                {
                    iCount1++;
                }   
            }
            else if ((ch == 'A') || (ch == 'Z'))
            {
                if(Arr[i] == ch)
                {
                    iCount2++;
                }   
            }
            
        }
        int Ans = iCount1 - iCount2;
       return Ans;

    }

    
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the String");
        String str = sobj.nextLine();

        System.out.println("Enter character");
        String ch = sobj.nextLine();

        char Arr[] = str.toCharArray();

        int iRet = FreqCharacter(Arr);
        System.out.println("Frequency of character : "+iRet);
    }
}